config/rbac/sg_role.yaml

```diff
@@ -0,0 +1,43 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: smart-gateway-role
+rules:
+  - apiGroups:
+      - authorization.k8s.io
+    resources:
+      - subjectaccessreviews
+    verbs:
+      - create
+  - apiGroups:
+      - authentication.k8s.io
+    resources:
+      - tokenreviews
+      - subjectaccessreviews
+    verbs:
+      - create
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: smart-gateway-role
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
```

Comment:
These look far too broad for the serviceaccount running the sg pod. I understand needing to add the SAR and tokenreviews for the oauth-proxy (Though I'm a bit concerned that I don't know what changed and it worked without them before). But where did the rest of these come from? They don't look relevant.

Reply:
I have dropped the scope of the RBAC down to what was suggested and it looks like the SGs are working. Will continue to watch, but I'm marking this resolved for now. Thanks!